runtimes/pythonrt/worker_grpc_handler.go

```diff
@@ -157,82 +161,71 @@ func (s *workerGRPCHandler) Activity(ctx context.Context, req *pb.ActivityReques
 
 // ak functions
 
-func makeCallbackMessage(args []sdktypes.Value, kwargs map[string]sdktypes.Value) *callbackMessage {
+func rpcSyscall[Resp proto.Message](
+	rid string,
+	callf func(context.Context, sdkservices.RunSyscalls) (sdktypes.Value, error),
+	errf func(error) Resp,
+	vf func(sdktypes.Value) Resp,
+) (Resp, error) {
+	w.mu.Lock()
+	runner, ok := w.runnerIDsToRuntime[rid]
+	w.mu.Unlock()
+	if !ok {
+		return errf(errors.New("runner id unknown")), nil
+	}
+
 	callbackChan := make(chan sdktypes.Value)
 	errorChannel := make(chan error)
 
-	msg := &callbackMessage{
-		args:           args,
-		kwargs:         kwargs,
+	runner.channels.callback <- &callbackMessage{
+		f:              callf,
 		successChannel: callbackChan,
 		errorChannel:   errorChannel,
 	}
-	return msg
+
+	select {
+	case err := <-errorChannel:
+		return errf(status.Errorf(codes.Internal, "syscall -> %v", err)), nil
+	case v := <-callbackChan:
+		return vf(v), nil
+	}
 }
 
 func (s *workerGRPCHandler) Sleep(ctx context.Context, req *pb.SleepRequest) (*pb.SleepResponse, error) {
 	if req.DurationMs < 0 {
 		return nil, status.Error(codes.InvalidArgument, "negative time")
 	}
 
-	w.mu.Lock()
-	runner, ok := w.runnerIDsToRuntime[req.RunnerId]
-	w.mu.Unlock()
-	if !ok {
-		return &pb.SleepResponse{
-			Error: "Unknown runner id",
-		}, nil
-	}
-
-	secs := float64(req.DurationMs) / 1000.0
-	args := []sdktypes.Value{
-		sdktypes.NewStringValue("sleep"),
-		sdktypes.NewFloatValue(secs),
-	}
-
-	msg := makeCallbackMessage(args, nil)
-
-	runner.channels.callback <- msg
-
-	select {
-	case err := <-msg.errorChannel:
-		err = status.Errorf(codes.Internal, "sleep(%f) -> %s", secs, err)
-		return &pb.SleepResponse{Error: err.Error()}, nil
-	case <-msg.successChannel:
-		return &pb.SleepResponse{}, nil
-	}
+	return rpcSyscall(
```

Comment:
This code is shorter but less explicit than the original version. I prefer more verbose and explicit.